client/login/wp-login/index.jsx

```diff
@@ -104,12 +104,16 @@ class Login extends React.Component {
 				</a>;
 		}
 
-		const goBackLink = ! magicLoginView && <a
-			href="#"
-			key="back-link"
-			onClick={ this.goBack }>
-				<Gridicon icon="arrow-left" size={ 18 } /> { this.props.translate( 'Return' ) }
-			</a>;
+		let goBackLink;
+		if ( typeof window === 'undefined' || window.history.length > 1 ) {
```

Comment:
Checkout the trick with `componentDidMount` [here](https://github.com/Automattic/delphin/blob/master/app/components/ui/form/email.js) and `state` here, maybe that's a better approach?